nir/src/main/scala/scala/scalanative/nir/Vals.scala

```diff
@@ -183,18 +262,37 @@ object Val {
       case _ => false
     }
   }
+
+  /** A heterogeneous collection of data members. */
   final case class StructValue(values: Seq[Val]) extends Val
+
+  /** A homogeneous collection of data members. */
   final case class ArrayValue(elemty: nir.Type, values: Seq[Val]) extends Val
+
+  /** A collection of bytes. */
   final case class ByteString(bytes: Array[scala.Byte]) extends Val {
     def byteCount: scala.Int = bytes.length + 1
   }
+
+  /** A local SSA variable. */
   final case class Local(id: nir.Local, valty: nir.Type) extends Val
+
+  /** A symbol. */
   final case class Global(name: nir.Global, valty: nir.Type) extends Val
 
-  // high-level
+  /** The unit value. */
   case object Unit extends Val
+
+  /** A constant. */
```

Comment:
Important information here is fact that `value` would be deduplicated in codegen and that the `Const` should be treated as pointer to a constant at the low level. It's not true that `Const` would work the same as literal constant - for these a `ByteString`, `Zero`, `Int`, `Long`, `Float`,  `StructValue`, `ArrayValue` etc can be used. These kind of vlaues are typically stored as a `value`